apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceHeaderPropagatingHandlerTest.cs

```diff
@@ -0,0 +1,144 @@
+            Assert.Empty(innerSpan.Labels);
+        }
+
+        [Fact]
+        public async Task TraceOutGoing_Exception()
+        {
+            string fakeUri = "http://www.thiscannotpossiblyexist934719238.com/";
+            string testId = Utils.GetTestId();
+            var spanName = $"{nameof(TraceOutGoing_Exception)}-{testId}";
+
+            var trace = await TestTracingOutGoingRequest(fakeUri, spanName, shouldCatch: true);
+            var innerSpan = trace.Spans.Single(s => s.Name != spanName);
+            var label = innerSpan.Labels.Single();
+            Assert.Equal(TraceLabels.StackTrace, label.Key);
+            Assert.False(string.IsNullOrWhiteSpace(label.Value));
+        }
+
+        /// <summary>
+        /// Helper funtion to test tracing of out going requests.
+        /// </summary>
+        /// <param name="uri">The uri to call.</param>
+        /// <param name="rootSpanName">The name for the root span.</param>
+        /// <param name="shouldCatch">True if exceptions from the request to the uri should be caught.</param>
+        /// <returns>A trace with two spans.  A root span and a child span tracing the outgoing uri
+        ///     request.</returns>
+        private async Task<TraceProto> TestTracingOutGoingRequest(
+            string uri, string rootSpanName, bool shouldCatch)
+        {
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+            await TraceOutGoingRequest(CreateTracer(), rootSpanName, uri, shouldCatch);
+            var trace = _polling.GetTrace(rootSpanName, startTime);
+            CheckTrace(trace, rootSpanName, uri);
+            return trace;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="SimpleManagedTracer"/> with a <see cref="GrpcTraceConsumer"/>.
+        /// </summary>
+        private SimpleManagedTracer CreateTracer()
+        {
+            string traceId = _traceIdFactory.NextId();
+            var traceProto = new TraceProto { ProjectId = _projectId, TraceId = traceId };
+            var consumer = new GrpcTraceConsumer(TraceServiceClient.CreateAsync());
+            return SimpleManagedTracer.Create(consumer, traceProto, null);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="TraceHeaderPropagatingHandler"/> and traces the sending of a
+        /// GET request to the given uri.  The trace is wrapped in a parent span.
+        /// </summary>
+        /// <param name="tracer">The tracer to trace the request with.</param>
+        /// <param name="rootSpanName">The name of the root span that will wrap the span
+        ///     that traces the request.</param>
+        /// <param name="uri">The uri to request.</param>
+        /// <param name="shouldCatch">True if exceptions from the request to the uri should be caught.</param>
+        private async Task TraceOutGoingRequest(
+            IManagedTracer tracer, string rootSpanName, string uri, bool shouldCatch)
```

Comment:
Actually, looking closer at how this is being used, it looks like you want as assertion failure for the test if `shouldCatch` is true but no exception is actually thrown by `await httpClient.GetAsync(uri)`, right? In that case, neither the original nor my rewritten code will accomplish this. You want something like this:

```csharp
try
{
    await httpClient.GetAsync(uri);
    Assert.False(shouldCatch);
}
catch (Exception e) when (shouldCatch && !(e is Xunit.Sdk.XunitException))
{
}
```

Also, since I think I have my head around what's going on here a bit better, I think `exceptionExpected` might be a better name for this parameter.